spec/controller/admin/codes_groups_controller_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+module Decidim
+  module AnonymousCodes
+    module Admin
+      describe CodeGroupsController, type: :controller do
+        routes { Decidim::AnonymousCodes::AdminEngine.routes }
+        let(:current_user) { create(:user, :confirmed, :admin, organization: current_organization) }
+        let(:current_organization) { create(:organization) }
+
+        before do
+          request.env["decidim.current_organization"] = current_organization
+          sign_in current_user, scope: :user
+        end
+
+        describe "GET #index" do
+          it "assigns @groups and renders the index template" do
+            group = Decidim::AnonymousCodes::Group.create(
+              title: "Sample Group",
+              expires_at: 1.day.from_now,
+              active: true,
+              max_reuses: 10,
+              organization: current_organization
+            )
```

Comment:
```suggestion
            group = create(:anonymous_codes_group, expires_at...)
```
You can also use factories here.
Although is better to define `group` as a `let`variable in the `describe` block:
```
describe "GET #index"
  let!(:group) { create(:anonymous_codes_group, organization: organization)
```

Reply:
Thanks, done.